### PR TITLE
New link checker workflow

### PR DESCRIPTION
**Pull request by dinakar29**
To make sure we're not adding dead, invalid or redirect links that CRAN typically checks.

### CODE REVIEW

**Comment by bms63** [comment]
Should it fall over if the links have errors in the summary?  I see a few of them are not relevant but not sure on the others.  Could these get placed into the exception part?

+ echo
## Errors per input
+ echo ::set-output name=exit_code::2
### Errors in .github/workflows/code-coverage.yml
+ '[' false = true ']'
* [irongut/CodeCoverageSummary@v1.2.0](irongut/CodeCoverageSummary@v1.2.0): Failed: Unreachable mail address: irongut/CodeCoverageSummary@v1.2.0: Invalid: Email doesn't exist or is syntactically incorrect

### Errors in _pkgdown.yml
* [https://github.com/pharmaverse/admiral/blob/main/](https://github.com/pharmaverse/admiral/blob/main/): Failed: Network error
```

### Errors in vignettes/writing_vignettes.Rmd
* [https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adxx.R](https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adxx.R): Failed: Network error

### Errors in .github/workflows/R-CMD-check.yml
* [https://packagemanager.rstudio.com/cran/__linux__/focal/latest](https://packagemanager.rstudio.com/cran/__linux__/focal/latest): Failed: Network error


```

**Comment by bms63** [comment]
Also...looks awesome and this will be a huge help for us.

**Comment by thomas-neitmann** [comment]
Great stuff, thanks a lot @dinakar29! Going bqck to what @bms63 said I would also expect the workflow to fail if any invalid links are detected. Otherwise I'm afraid people will never bother to take a look if the workflow gets a green check mark every time.

**Comment by dinakar29** [comment]
Yeah this is a WIP. Standby for more updates.

**Comment by dinakar29** [comment]
@bms63 and @thomas-neitmann How's that? A PR comment posted by actions with the summary.

**Comment by bms63** [comment]
Is there a way to add those invalid links to the exceptions?  Can we do that ourselves? 

Can errors be kept somewhere else?  It is a lot of information.  If the link check fails then people can go look at what is the root cause.  We have this and the code coverage report so I feel like folks might get overwhelmed!  

Very excited for this addition to our actions!!!!!!

**Comment by bms63** [comment]
Did the bot get updated?

**Comment by dinakar29** [comment]
> Did the bot get updated?

Which bot?

**Comment by bms63** [comment]
the link check bot?

**Comment by dinakar29** [comment]
> the link check bot?

There's not bot. It's a job. Unless you're referring to the Github Actions bot which won't post anything on behalf of the job since I disabled that at your behest.

**Comment by bms63** [comment]
> > the link check bot?
> 
> There's not bot. It's a job. Unless you're referring to the Github Actions bot which won't post anything on behalf of the job since I disabled that at your behest.

Ah...So do we need to test this out more?  Do I add the lycheeignore file or will you add it?

**Comment by dinakar29** [comment]
I wouldn't know what's valid and what's not (from a maintenance perspective) - so I'd say go ahead and add the `.lycheeignore` with the URLs you want to skip scans for. More info on how to use it here: https://github.com/lycheeverse/lychee#ignoring-links

Note that CRAN may not ignore the same links that we ignore in our checks, so it's best to fist remediate anything that CRAN reports and then add whatever it is we want to ignore (stuff that CRAN doesn't bother checking like GH Action workflow files etc) to the ignore file.

**Comment by dinakar29** [comment]
@bms63 - you need to place `.lycheeignore` at the *root* of the project, not in `.github/workflows/`

File: vignettes/admiral.Rmd [hunk 1]
```diff
@@ -283,6 +283,6 @@ channel](https://app.slack.com/client/T028PB489D3/C02M8KN8269).
 
 # See also
 
-- [Template scripts](https://github.com/pharmaverse/admiral/tree/main/inst/templates)
+- [Template scripts](https://github.com/pharmaverse/admiral/tree/main/inst/templatesxyz)
```
Comment (bms63) [inline vignettes/admiral.Rmd:286]: but does flag this one

Comment (dinakar29) [reply vignettes/admiral.Rmd:286]: The string matches a valid URL regex, but the URL itself is inaccessible, hence it's flagged

File: vignettes/admiral.Rmd [hunk 2]
```diff
@@ -258,7 +258,7 @@ for the most common steps are provided by the following vignettes:
 
 - [Creating ADSL](adsl.html)
 - [Creating an OCCDS ADaM](occds.html)
-- [Creating a BDS Finding ADaM](bds_finding.html)
+- [Creating a BDS Finding ADaM](bds_finding.htmlxyz)
```
[the remark this answers is not in the document]
Comment (dinakar29) [reply vignettes/admiral.Rmd:261]: Yeah exactly. Relative links in markdown are hard to validate unless they are rendered to their final HTML docs.

[the remark this answers is not in the document]
Comment (bms63) [reply vignettes/admiral.Rmd:261]: Should we just do absolute links?  I feel like there was  reason why we started doing relative links, but can't remember.

[the remark this answers is not in the document]
Comment (dinakar29) [reply vignettes/admiral.Rmd:261]: No absolute links are a bad idea in markdown. Leave these as-is. Don't need to overfit.

**Pull request merged**